validate/validate.go

```diff
@@ -100,7 +100,9 @@ func (v *Validator) CheckAll() (msgs []string) {
 	msgs = append(msgs, v.CheckMandatoryFields()...)
 	msgs = append(msgs, v.CheckSemVer()...)
 	msgs = append(msgs, v.CheckMounts()...)
-	msgs = append(msgs, v.CheckProcess()...)
+	if v.spec.Process != nil {
+		msgs = append(msgs, v.CheckProcess()...)
```

Comment:
`Hooks` is a similar pointer, and we're currently handling that by [unconditionally adding `CheckHooks`][1] and then [guarding against the `nil` inside `CheckHooks`][2].  I don't have a strong preference about whether the nil guard is inside `CheckAll` or the more specific `Check*` method, but I think we want to stay consistent.

[1]: https://github.com/opencontainers/runtime-tools/blob/2d270b8764c02228eeb13e36f076f5ce6f2e3591/validate/validate.go#L104
[2]: https://github.com/opencontainers/runtime-tools/blob/2d270b8764c02228eeb13e36f076f5ce6f2e3591/validate/validate.go#L175